go/obscurocommon/common.go

```diff
@@ -23,6 +23,6 @@ type Nonce = uint64
 type EncodedRollup []byte
 
 type NotifyNewBlock interface {
-	MockedNewHead(b EncodedBlock, p EncodedBlock)
+	MockedNewHead(b EncodedBlock)
```

Comment:
this won't work. It will introduce a lot of flakyness, unless we replace it with a better mechanism.
The reason for always sending the parent with every block is to avoid races where the child arrives before the parent , and then things blow up

Reply:
Makes sense.